experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -0,0 +1,654 @@
+        M = matrix(ring,m,m,[0 for i in 1:m^2])
+        for i in 1:m
+            M[1:m,i] = gens(ring)[m*(i-1)+1:i*m]
+        end
+        G.canonical_representation = M
+        G.group_ideal = ideal([det(M) - 1])
+        #base ring of M has to be the same as the representation matrix when that is created later.
+        return G
+    end
+end
+
+function Base.show(io::IO, G::ReductiveGroup)
+    print(io, "Reductive group ", G.group[1], G.group[2])
+end
+
+function reductive_group(sym::Symbol, m::Int, F::Field)
+    return ReductiveGroup(sym,m,F)
+end
+
+function reductive_group(sym::Symbol, m::Int, R::MPolyRing)
+    return ReductiveGroup(sym,m,R)
+end
+
+group(G::ReductiveGroup) = G.group
+reynolds_operator(G::ReductiveGroup) = G.reynolds_operator
+group_ideal(G::ReductiveGroup) = G.group_ideal
+canonical_representation(G::ReductiveGroup) = G.canonical_representation
+natural_representation(G::ReductiveGroup) = G.canonical_representation
+
+#####################
+#Representation
+#####################
+
+mutable struct RepresentationReductiveGroup
+    group::ReductiveGroup
+    rep_mat::AbstractAlgebra.Generic.MatSpaceElem
+    sym_deg::Tuple{Bool, Int}
+    reynolds_v::Function
+    
+    #representation of group G over symmetric degree d
+    function RepresentationReductiveGroup(G::ReductiveGroup, d::Int)
+        R = new()
+        R.group = G
+        R.rep_mat = rep_mat_(G, d, G.field)
+        R.sym_deg = (true, d)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+    
+    #does not check the given matrix.
+    function RepresentationReductiveGroup(G::ReductiveGroup, M::AbstractAlgebra.Generic.MatSpaceElem)
+        base_ring(M) == base_ring(G.group_ideal) || error("Group ideal and representation matrix must have same parent ring")
+        R = new()
+        R.group = G
+        R.rep_mat = M
+        R.sym_deg = (false, 0)
+        R.reynolds_v = reynolds_v_slm
+        return R
+    end
+end
+
+function representation_reductive_group(G::ReductiveGroup, d::Int)
+    return RepresentationReductiveGroup(G, d)
+end
+
+representation_matrix(R::RepresentationReductiveGroup) = R.rep_mat
+reductive_group(R::RepresentationReductiveGroup) = R.group
+function vector_space_dimension(R::RepresentationReductiveGroup)
+    return ncols(R.rep_mat)
+end
+
+function Base.show(io::IO, R::RepresentationReductiveGroup)
+    println(io, "Representation of ", R.group.group[1], R.group.group[2])
+    if R.sym_deg[1]
+        print(io, "over symmetric forms of degree ", R.sym_deg[2])
+    else
+        println(io, "with representation matrix")
+        show(io, R.rep_mat)
+    end
+end
+
+###############
+
+#for the representation matrices
+
+function rep_mat_(G::ReductiveGroup, sym_deg::Int, F::Field)
+    G.group[1] == :SL || error("Only implemented for SLm")
+    m = G.group[2]
+    n = binomial(m + sym_deg - 1, m - 1)
+    mixed_ring, t, z, a = PolynomialRing(F, "t"=> 1:m, "z"=> (1:m, 1:m), "a" => 1:n)
```

Comment:
We don't need the `a` anymore, do we?
```suggestion
    mixed_ring, t, z = polynomial_ring(F, "t"=> 1:m, "z"=> (1:m, 1:m))
```